server/command.go

```diff
@@ -234,15 +234,33 @@ func (p *Plugin) executeUnlinkCommand(args *model.CommandArgs) (*model.CommandRe
 		return p.cmdError(args.UserId, args.ChannelId, "Unable to unlink the channel, you have to be a channel admin to unlink it.")
 	}
 
-	if _, err := p.store.GetLinkByChannelID(channel.Id); err != nil {
+	link, err := p.store.GetLinkByChannelID(channel.Id)
+	if err != nil {
+		p.API.LogDebug("Unable to get the link by channel ID", "error", err.Error())
 		return p.cmdError(args.UserId, args.ChannelId, "This Mattermost channel is not linked to any MS Teams channel.")
 	}
 
-	if err := p.store.DeleteLinkByChannelID(channel.Id); err != nil {
+	if err = p.store.DeleteLinkByChannelID(channel.Id); err != nil {
 		return p.cmdError(args.UserId, args.ChannelId, "Unable to delete link.")
 	}
 
 	p.sendBotEphemeralPost(args.UserId, args.ChannelId, "The MS Teams channel is no longer linked to this Mattermost channel.")
+
+	subscription, err := p.store.GetChannelSubscriptionByTeamsChannelID(link.MSTeamsChannel)
+	if err != nil {
+		p.API.LogDebug("Unable to get the subscription by MS Teams channel ID", "error", err.Error())
+		return &model.CommandResponse{}, nil
+	}
+
+	if err = p.store.DeleteSubscription(subscription.SubscriptionID); err != nil {
+		p.API.LogDebug("Unable to delete the subscription from the DB", "subscriptionID", subscription.SubscriptionID, "error", err.Error())
+		return &model.CommandResponse{}, nil
+	}
+
+	if err = p.msteamsAppClient.DeleteSubscription(subscription.SubscriptionID); err != nil {
```

Comment:
i think we should change the order to delete things here. Something like deleting the subscription from MS Teams followed by deleting the subscription from db and then deleting the link itself.

Reply:
Nope, this order was decided after thinking for a long time. We can discuss this on a call.